### PR TITLE
Champagne poppin.

### PR DESCRIPTION
**Pull request by Krysonism**
## About The Pull Request

This PR makes champagne bottles a little more realistic.

Unlike most other bottles, you now have to open them first, this of course sends the cork flying and produces a satisfying popping sound.
The cork itself has 10 throwforce and when it escapes the bottle it is imbued with a little know mystical force that causes it to knockdown for 2 seconds.

I have also polished up the sprite ever so slightly, it was the first SS13 sprite i ever made, so I could not bear to resprite it fully despite it being pretty soulful.
 

## Why It's Good For The Game
I have an emotional attachment to the champagne bottle, but it really isn't that interesting in its current state.

It is almost a pure RP item and an expensive one at that, so i think it aleast deserves to have some flair. 

## Changelog
:cl:
expansion: You now need to first uncork the bottle before enjoying your champagne./:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
We are one step closer to recreating that LMFAO music video where they pop holy champagne and kill vampires with jesus christ

**Comment by maxymax13** [comment]
does the cork fly in any particular direction? can you aim it to knock the clown off the tram rails and make it look like an accident?

File: code/modules/food_and_drinks/drinks/drinks/bottle.dm [hunk 1]
```diff
@@ -485,9 +485,56 @@
...
+		icon_state = initial(icon_state)
+
+/obj/item/reagent_containers/food/drinks/bottle/champagne/proc/pop_cork(mob/user)
+	user.visible_message(span_danger("[user] loosens the cork of [src] causing it to pop out of the bottle with great force."), \
+				span_nicegreen("You elegantly loosen the cork of [src] causing it to pop out of the bottle with great force."))
+	reagents.flags |= OPENCONTAINER
+	playsound(src, 'sound/items/champagne_pop.ogg', 70, TRUE)
+	spillable = TRUE
+	update_appearance()
+	var/obj/item/champagne_cork/popped_cork = new(get_turf(src))
```
Comment (Kylerace) [inline code/modules/food_and_drinks/drinks/drinks/bottle.dm:515]: should just be loc otherwise you could hit people with corks inside of containers

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drinks/drinks/drinks/bottle.dm [hunk 2]
```diff
@@ -485,9 +485,56 @@
...
+	reagent_flags = TRANSPARENT
+	spillable = FALSE
 	isGlass = TRUE
 	list_reagents = list(/datum/reagent/consumable/ethanol/champagne = 100)
 
+
+/obj/item/reagent_containers/food/drinks/bottle/champagne/attack_self(mob/user)
+	if(spillable)
+		return ..()
+	to_chat(user, span_notice("You fiddle with the cork of [src]."))
```
Comment (Mothblocks) [inline code/modules/food_and_drinks/drinks/drinks/bottle.dm:497]:
```suggestion
	balloon_alert(user, "fiddling with cork...")
```

File: code/modules/food_and_drinks/drinks/drinks/bottle.dm [hunk 3]
```diff
@@ -485,9 +485,56 @@
...
+/obj/item/reagent_containers/food/drinks/bottle/champagne/update_icon_state()
+	. = ..()
+	if(spillable)
+		icon_state = "[initial(icon_state)]_popped"
+	else
+		icon_state = initial(icon_state)
+
+/obj/item/reagent_containers/food/drinks/bottle/champagne/proc/pop_cork(mob/user)
+	user.visible_message(span_danger("[user] loosens the cork of [src] causing it to pop out of the bottle with great force."), \
+				span_nicegreen("You elegantly loosen the cork of [src] causing it to pop out of the bottle with great force."))
```
Comment (Mothblocks) [inline code/modules/food_and_drinks/drinks/drinks/bottle.dm:510]:
```suggestion
		span_nicegreen("You elegantly loosen the cork of [src] causing it to pop out of the bottle with great force."))
```

You should never be indenting more than once.

File: code/modules/food_and_drinks/drinks/drinks/bottle.dm [hunk 4]
```diff
@@ -485,9 +485,56 @@
...
+	if(spillable)
+		return ..()
+	to_chat(user, span_notice("You fiddle with the cork of [src]."))
+	if(do_after(user, 1 SECONDS, src))
+		return pop_cork(user)
+
+/obj/item/reagent_containers/food/drinks/bottle/champagne/update_icon_state()
+	. = ..()
+	if(spillable)
+		icon_state = "[initial(icon_state)]_popped"
```
Comment (Mothblocks) [inline code/modules/food_and_drinks/drinks/drinks/bottle.dm:504]: Use `base_icon_state` so admins can mess with it.

File: code/modules/food_and_drinks/drinks/drinks/bottle.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/food_and_drinks/drinks/drinks/bottle.dm:515]: +1

File: code/modules/food_and_drinks/drinks/drinks/bottle.dm [hunk 5]
```diff
@@ -485,9 +485,56 @@
...
+	popped_cork.throw_at(target = pop_target, thrower = user, speed = 2, range = 7, spin = FALSE)
+
+/obj/item/champagne_cork
+	name = "champagne cork"
+	icon = 'icons/obj/drinks.dmi'
+	icon_state = "champagne_cork"
+	force = 0
+	throwforce = 10
+	///this var determines if the cork will knockdown on impact.
+	var/might_of_the_sun_king = FALSE
```
Comment (Mothblocks) [inline code/modules/food_and_drinks/drinks/drinks/bottle.dm:527]: ill let you have this just this once

File: sound/attributions.txt [hunk 6]
```diff
@@ -17,4 +17,5 @@ captured by the BBC
 "Gunfire & Guns: Revolver, Chambers spinning.": http://bbcsfx.acropolis.org.uk/assets/07027165.wav
 bbc.co.uk - (c) copyright 2018 BBC
 
-kiss sound: https://freesound.org/people/stereostereo/sounds/117355/
\ No newline at end of file
+kiss sound: https://freesound.org/people/stereostereo/sounds/117355/
+champagne_pop.ogg is credited to ultradust on freesound https://freesound.org/people/ultradust/sounds/166923/
```
Comment (Mothblocks) [inline sound/attributions.txt:21]:
```suggestion
champagne_pop.ogg is credited to ultradust on freesound https://freesound.org/people/ultradust/sounds/166923/

```

New line prevents blames from getting worse, just like this diff shows with the kiss sound.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Krysonism** [comment]
I will address all your concerns tonight

**Comment by Krysonism** [comment]
Ended up working overtime at the factory and then fell asleep for 12 hours.

**Comment by Mothblocks** [comment]
no rush

**Review by Twaticus — APPROVED** [review]

File: sound/attributions.txt (hunk 6 above)
Comment (Mothblocks) [inline sound/attributions.txt:21]:
```suggestion
champagne_pop.ogg is credited to ultradust on freesound https://freesound.org/people/ultradust/sounds/166923/

```

**Comment by Krysonism** [comment]
Thanks, i forgot that one.

**Pull request merged**